frontend/src/components/activity/content/Checklist.vue

```diff
@@ -0,0 +1,260 @@
+    return {
+      savingRequestCount: 0,
+      dirty: false,
+      showDialog: false,
+      checkedItems: [],
+      uncheckedItems: [],
+      errorMessages: null,
+      debouncedSave: () => null,
+      itemsLoaded: false,
+    }
+  },
+  computed: {
+    campChecklistItems() {
+      return this.api.get().checklistItems().items
+    },
+    selectionContentNode() {
+      return this.campChecklistItems.filter((item) =>
+        this.contentNode
+          .checklistItems()
+          .items.some(({ _meta }) => _meta.self === item._meta.self)
+      )
+    },
+    serverSelection() {
+      return this.selectionContentNode.map((item) => item.id)
+    },
+    allChecklists() {
+      return this.camp.checklists().items.map((checklist) => ({
+        checklist,
+        items: this.campChecklistItems
+          .filter((item) => item.checklist()._meta.self === checklist?._meta.self)
+          .map((item) => ({
+            item,
+            parents: this.itemsLoaded ? this.getParents(item) : [],
+          }))
+          .sort((a, b) => {
+            const aparents = [
+              ...a.parents.map(({ position }) => position),
+              a.item.position,
+              -1,
+            ]
+            const bparents = [
+              ...b.parents.map(({ position }) => position),
+              b.item.position,
+              -1,
+            ]
+            for (let i = 0; i < Math.min(aparents.length, bparents.length); i++) {
+              if (aparents[i] !== bparents[i]) {
+                return aparents[i] - bparents[i]
+              }
+            }
+            return 0
+          }),
+      }))
+    },
+    activeChecklists() {
+      return this.allChecklists
+        .filter(({ checklist }) =>
+          this.contentNode
+            .checklistItems()
+            .items.some((item) => checklist._meta.self === item?.checklist()._meta.self)
+        )
+        .map(({ checklist, items }) => ({
+          checklist,
+          items: items.filter(({ item }) => this.checkedItems.includes(item.id)),
+        }))
+    },
+  },
+  watch: {
+    serverSelection: {
+      async handler(newOptions, oldOptions) {
+        if (isEqual(sortBy(newOptions), sortBy(oldOptions))) {
+          return
+        }
+
+        // copy incoming data if not dirty or if incoming data is the same as local data
+        if (!this.dirty || isEqual(sortBy(newOptions), sortBy(this.checkedItems))) {
+          this.resetLocalData()
+        }
+      },
+      immediate: true,
+    },
+  },
+  created() {
+    const DEBOUNCE_WAIT = 500
+    this.debounceSave = debounce(this.save, DEBOUNCE_WAIT)
+    this.api
+      .get()
+      .checklistItems()
+      ._meta.load.then(() => {
+        this.itemsLoaded = true
+      })
+  },
+  beforeDestroy() {
+    this.checkedItems = null
+    this.uncheckedItems = null
+  },
+  methods: {
+    // get all parents of a given item
+    getParents(item) {
+      const parents = []
+      let parent = item?.parent?.()
```

Comment:
what is happending here?

Reply:
I need all parents to be able to display the path